### PR TITLE
[entropy] audit 4. users can influence the Entropy revealed result

### PR DESCRIPTION
**Pull request by 0xfirefist**

### CODE REVIEW

File: target_chains/ethereum/contracts/contracts/entropy/Entropy.sol
Comment (jayantk) [inline target_chains/ethereum/contracts/contracts/entropy/Entropy.sol:261]: this comment is confusing. I don't know what a "stale request" is, and this comment assumes I do. Then it proceeds to say "well also it fails when the blockhash is 0 for a different reason".

I think the comment should explain why the blockhash can be zero, and why we need this check in those cases.

Also suggest changing the error to be `BlockhashUnavailable`, which is a more precise description of the problem.

File: target_chains/ethereum/contracts/forge-test/Entropy.t.sol
Comment (jayantk) [inline target_chains/ethereum/contracts/forge-test/Entropy.t.sol:408]: why is this 1236 and not 1235? I would have thought 1235 would work too.

File: target_chains/ethereum/contracts/forge-test/Entropy.t.sol
Comment (jayantk) [inline target_chains/ethereum/contracts/forge-test/Entropy.t.sol:496]: I think this test would be better if there was exactly 1 `sequenceNumber` generated at the beginning, then you test the two revert conditions, then you test that the reveal succeeds afterward.

That way of writing this test is less error prone, because it ensures the block number is the only variable that is changing between the reveal calls.

**Review by jayantk — APPROVED** [review]
please address the comments before merging. They're easy though so no need to get another review from me.

File: target_chains/ethereum/contracts/forge-test/Entropy.t.sol
Comment (0xfirefist) [reply target_chains/ethereum/contracts/forge-test/Entropy.t.sol:408]: no reason as such.

File: target_chains/ethereum/contracts/forge-test/Entropy.t.sol
Comment (0xfirefist) [reply target_chains/ethereum/contracts/forge-test/Entropy.t.sol]: oh yeah!

**Pull request merged**
